src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,264 @@
+import React, {useMemo, useRef, useState} from 'react';
+import {
+  Animated,
+  TextInput as InputField,
+  NativeSyntheticEvent,
+  StyleSheet,
+  Text,
+  TextInputFocusEventData,
+  View,
+} from 'react-native';
+import {fontStyles} from '../../constants';
+import {useThemeColor} from '../../hooks';
+import {Strings} from '../../i18n';
+import {Label} from '../Label';
+import {TextInputProps} from './TextInput.types';
+import {horizontalScale, verticalScale} from '../../utils';
+
+export const TextInput = (props: TextInputProps) => {
+  const {
+    mode = 'default',
+    onChangeText,
+    hint,
+    style,
+    placeholder,
+    hintStyle,
+    error,
+    errorMessage,
+    errorMessageStyle,
+    label,
+    labelStyle,
+    editable,
+    inputStyle,
+    left,
+    right,
+    value,
+    ...restProps
+  } = props;
+  const [focused, setFocused] = useState(false);
+  const {colors} = useThemeColor();
+
+  const getStyle = useMemo(() => {
+    if (mode === 'default') {
```

Comment:
Why add this check ? We can just return at the end if mode was not outline or border